### ISSUE TITLE
Regression: Toolbar overflow menu no longer scrollable

### ISSUE THREAD

**Issue by ThomasH99** [codenameone/CodenameOne#3584]
**Describe the bug**
Sometime between March 20 this year and today, the scrolling of Toolbar overflow menu has stopped working. This is in particular an issue in landscape mode where a menu with more than a few options becomes inaccessible. 

It actually seems that the menu *does* scroll, but invisibly: in my app, if I try to scroll the menu in landscape, it does not scroll visibly, but then when I click a menu command it launches *another* menu command (which is further down the list of overflow commands). 

**To Reproduce**
Run this example, in Simulator or on device (iOS). The overflow menu won't scroll (visibly) if too long to be displayed. 
```
Form hi = new Form("No menu scroll in landscape", BoxLayout.y());
Toolbar t = hi.getToolbar();
for (int i = 0; i < 25; i++) {
    t.addCommandToOverflowMenu(Command.createMaterial("Menu-" + i, FontImage.MATERIAL_ADD_COMMENT, null));
}
hi.add(new Button("XXXXXXX"));
hi.show();

```
**Expected behavior**
The overflow menu should scroll like before.

**Comment by AnthonyRivers** [comment]
Hi ThomasH99, yes the the scroll does not work in the Toolbar overflow menu I was able to replicated as well. I think the problem is that the overflow menu is using a Dialog which doesn't have setScrollableY set to true.  I'm just a newbie with CodenameOne I've been trying to learn the API for the past 4 weeks, but couldn't you just make your own overflow menu? Trying to trace where this bug is in the source actually helped me understand the code in a deeper level thanks for reporting this problem.  I'm no expert, but I was able to emulate the overflow menu which scrolls. It's not as slick and good looking as the built in one, but I was actually pretty happy I figure out how to replicate the overflow menu.

![image](https://user-images.githubusercontent.com/4259437/165880011-633af632-23c6-469c-899c-2e23d6b0b56e.png)

I think the problem in the source code in the Toolbar implementation when the sideMenu.show() is called in line 2627.
![image](https://user-images.githubusercontent.com/4259437/165880396-0ccc576a-30fe-4310-9ed1-f240cff45f7a.png)

If you see the Dialog used for the sideMenu has setScrollable set to false in line 704.
![image](https://user-images.githubusercontent.com/4259437/165880550-b9e96902-1e63-48f1-a8ef-90e2357d6987.png)

I spent like to days trying to find a way to access the dialog, but I was not able to find a way to set the scrollable value for the Dialog used for the overflow menu. The component inspector in the simulator doesn't even show the dialog being used. It only shows the Button. So I don't think we can set scrollable freely or access the dialog used for the overflow menu at all.

![image](https://user-images.githubusercontent.com/4259437/165880856-0f1315bf-9e06-423a-a97a-26acfa364da2.png)

**Comment by ThomasH99** [comment]
Thanks @AnthonyRivers. So, it sounds like a regression to me that should be reversed in CN1 (I don't think it makes sense to have the menu non-scrollable, there are only downsides). @shai-almog something you will look at?

**Comment by ThomasH99** [comment]
I've believe I found the issue. I investigated a bit more and did not find any recent changes in `Toolbar` or `MenuBar` that would explain the regression. The reason seems to be the below recent change in Dialog. At least, when I comment out the `&& CN.getCurrentForm() == this` the overflow menu scrolls normally again. Not obvious to me why the test was added. Could @shai-almog  or @shannah clarify and possibly fix the issue? 
```
  void repaint(Component cmp) {
        if(getParent() != null){
            super.repaint(cmp);
            return;
        }
        if (isVisible() && !disposed && CN.getCurrentForm() == this) { //<-- the addition of "&& CN.getCurrentForm() == this" prevents an Overflow Menu from updating/repainting when scrolling
            Display.getInstance().repaint(cmp);
        }
    }
```

**Comment by ThomasH99** [comment]
Do you approve this fix? Do I need to make PR to move this forward?

**Comment by shannah** [comment]
This will be fixed in next update (7.0.73).
